packages/dev/core/src/Meshes/meshUVSpaceRenderer.ts

```diff
@@ -145,36 +195,152 @@ export class MeshUVSpaceRenderer {
      * @param angle The rotation angle around the direction of the projection
      */
     public renderTexture(texture: BaseTexture, position: Vector3, normal: Vector3, size: Vector3, angle = 0): void {
+        // Create the diffuse render target texture if it doesn't exist
+        if (!this.texture && !this._options.uvEdgeBlending && !this._decalTexture) {
+            this._updateRTT();
+        } else if (this.texture && !this._userCreatedTextureSetup && !this._options.uvEdgeBlending && !this._decalTexture) {
+            this._updateRTT();
+        } else if (this.texture && !this._userCreatedTextureSetup && this._options.uvEdgeBlending && !this._decalTexture) {
+            this._createDecalDiffuseRTT();
+        } else if (!this.texture && this._options.uvEdgeBlending && !this._decalTexture) {
+            this._createDecalDiffuseRTT();
+        }
+
+        // // Prepare the shader with the decal texture, mask texture, and necessary uniforms
+        const shader = MeshUVSpaceRenderer._GetShader(this._scene);
+        shader.setTexture("textureSampler", texture); // Decal texture
+
+        // Calculate and set the projection matrix
+        const projectionMatrix = this._createProjectionMatrix(position, normal, size, angle);
+        shader.setMatrix("projMatrix", projectionMatrix);
+
+        if (!this._options.uvEdgeBlending) {
+            if (this.texture instanceof RenderTargetTexture) {

```

Comment:
Can you use MeshUVSpaceRenderer._isRenderTargetTexture instead for perf reasons ?